genet/utils/google_directions.py

```diff
@@ -67,18 +69,43 @@ def read_saved_api_results(file_path):
     return api_requests
 
 
-def make_request(origin_attributes, destination_attributes, key, traffic):
+def make_request(origin_attributes, destination_attributes, key, departure_time, traffic_model):
     base_url = 'https://maps.googleapis.com/maps/api/directions/json'
-    params = {
-        'origin': '{},{}'.format(origin_attributes['lat'], origin_attributes['lon']),
-        'destination': '{},{}'.format(destination_attributes['lat'], destination_attributes['lon']),
-        'key': key
-    }
-    if traffic:
+    params = {'origin': '{},{}'.format(origin_attributes['lat'], origin_attributes['lon']),
+              'destination': '{},{}'.format(destination_attributes['lat'], destination_attributes['lon']), 'key': key,
+              'traffic_model': traffic_model}
+    current_unix_time = time.time()
+    if departure_time == -1:
         params['departure_time'] = 'now'
+    elif (type(departure_time) == int) & (departure_time > current_unix_time) & (departure_time < 2147483646):
```

Comment:
i looks like it's the end of time.. unix time 😂  http://unixtimestamps.rsmith.co/unixtimestamp/2147483647

>What happens on January 19, 2038?
In early 2038 the Unix Time Stamp, if uses 32-bit signed integer, will overflow. This is really bad. Older systems will need to upgrade their timestamp storage to cope to avoid failure. Read more on wikipedia.

Reply:
I have made it into a constant `end_of_unix_time`